src/lib/coda_base/staged_ledger_hash.ml

```diff
@@ -41,16 +41,144 @@ module Aux_hash = struct
   let to_bytes = Fn.id
 
   let dummy : t = String.init length_in_bytes ~f:(fun _ -> '\000')
+end
+
+module Pending_coinbase_extra = struct
+  let length_in_bits = 256
+
+  let length_in_bytes = length_in_bits / 8
+
+  module Stable = struct
+    module V1 = struct
+      module T = struct
+        let version = 1
+
+        type t = string [@@deriving bin_io, sexp, eq, compare, hash]
+      end
+
+      include T
+      include Registration.Make_latest_version (T)
+    end
+
+    module Latest = V1
+
+    module Module_decl = struct
+      let name = "staged_ledger_hash_pending_coinbase_state_hash"
+
+      type latest = Latest.t
+    end
+
+    module Registrar = Registration.Make (Module_decl)
+    module Registered_V1 = Registrar.Register (V1)
+  end
+
+  include Stable.Latest
+
+  let of_bytes = Fn.id
+
+  let to_bytes = Fn.id
+
+  let dummy : t = String.init length_in_bytes ~f:(fun _ -> '\000')
+end
+
+module Non_snark = struct
```

Comment:
Let's see if we can come up with a better name

Reply:
`Unchecked_hash`?